crates/runtime/src/sources/mod.rs

```diff
@@ -0,0 +1,60 @@
+//! Sources for Vixen.
+//!
+//! A `Source` is a trait that defines the behavior for data `Sources` that can be used to connect to it and
+//!  send the updates to a channel.
+//!
+//! The `Source` trait is implemented by the `yellowstone_grpc` module.
+
+use async_trait::async_trait;
+use tokio::sync::mpsc::Sender;
+use vixen_core::Filters;
+use yellowstone_grpc_proto::{geyser::SubscribeUpdate, tonic::Status};
+
+use crate::config::YellowstoneConfig;
+
+/// A `Source` implementation for the Yellowstone gRPC API.
+pub mod yellowstone_grpc;
+
+/// A `Source` implementation for Solana Accounts backfilling through the RPC API.
+pub mod solana_accounts_rpc;
```

Comment:
How much does the dependency tree change if we split out sources into its own crate or a crate per source. There is value if vixen runtime doesn't share many crates with sources.

Ideally runtime has no dependencies on any sort of specific source. And each source crate doesn't share crates with each other except for `yellowstone-vixen-sources`